fix: Missing focus management when on PDP 'Show Reviews' is clicked (#14030)

pawelfras opened the pull request:
closes #14030

pawelfras committed:
fix: Missing focus management when on PDP 'Show Reviews' is clicked (#14030)

developpeurweb reviewed (dismissed):
Works well on JAWS/NVDA/VO.
Pls make sure an angular dev reviews the code before merging

FollowTheFlo reviewed (dismissed):
tested all scenario i could think: switch tabs/accordion view, resize, voice over.
ran unit and e2e tests.
Code looks good to me, no enhancements or grey area i could think.

pawelfras committed:
small refactor

pawelfras committed:
Merge branch 'develop' into feature/GH-14030

pawelfras committed:
small update in changes desc

pawelfras committed:
handling SSR

Platonn commented on projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts:
```diff
@@ -2,59 +2,106 @@ import {
-  // but part of tabs component. This is likely to change in refactor.
-  private getReviewsComponent(): Element {
-    return this.winRef.document.querySelector('cx-product-reviews');
+  // Create a mutation observer to observe if reviews component has been rendered
```
Let's turn it to JS doc comment so customers may benefit from it.

pawelfras committed:
prettier fix

pawelfras committed:
refactor

pawelfras committed:
Merge branch 'develop' into feature/GH-14030

BenjaminEmiliani commented on projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts:
```diff
@@ -13,52 +14,84 @@ import { CurrentProductService } from '../current-product.service';
 })
 export class ProductIntroComponent implements AfterContentChecked {
-  reviewsTabAvailable = new BehaviorSubject<boolean>(false);
+  product$: Observable<Product | null> =
```
Hey @pawelfras I wonder if this should be added to 5.0md? Here from documentation it says functions but I don't see why it would not also apply to public properties
<img width="440" alt="Screen Shot 2022-03-18 at 9 56 55 AM" src="https://user-images.githubusercontent.com/49131189/159016976-5a1b2640-b1c8-4430-a1bb-e071127f9287.png">

pawelfras committed:
refactor and change notes

pawelfras committed:
Merge branch 'develop' into feature/GH-14030

pawelfras committed:
Merge branch 'develop' into feature/GH-14030

BenjaminEmiliani commented on projects/storefrontlib/cms-components/product/product-intro/product-intro.component.spec.ts:
```diff
@@ -34,20 +45,23 @@ describe('ProductIntroComponent in product', () => {
 
-  it('should be created', () => {
-    expect(productIntroComponent).toBeTruthy();
-  });
```
Hey @pawelfras is there a reason why you removed this spec?

pawelfras replied:
I was always wondering about value added of this kind of tests, but this is a discussion for another topic. Test has been reverted to keep consistency.

BenjaminEmiliani commented on projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts:
```diff
@@ -13,52 +13,68 @@ import { CurrentProductService } from '../current-product.service';
+          setTimeout(() => {
+            reviewsTab.scrollIntoView({ behavior: 'smooth' });
+            reviewsTab.focus({ preventScroll: true });
+          }, 0);
```
Very minor note but if timeout is 0, we can omit the parameter

BenjaminEmiliani commented on docs/migration/5_0.md:
```diff
@@ -498,14 +498,23 @@ The following keys have been added to `configurator-common.ts`:
+
+### ProductIntroComponent
+
+- Subject `reviewsTabAvailable` has been renamed to "areReviewsAvailable$".
```
@pawelfras I think this doc note should be:
Property \`reviewsTabAvailable\` has been renamed to \`areReviewsAvailable$\`.

pawelfras committed:
refactor after review

pawelfras committed:
refactor

pawelfras committed:
change notes

pawelfras committed:
Merge branch 'develop' into feature/GH-14030

pawelfras committed:
revert e2e changes

developpeurweb reviewed (dismissed):
Still works well on JAWS/NVDA/VO.

pawelfras committed:
feat: E2E and stability improvement for Show reviews button (#15430)

pawelfras committed:
Merge branch 'develop' into feature/GH-14030

pawelfras committed:
Merge branch 'develop' into feature/GH-14030

pawelfras committed:
Trigger Build

pawelfras committed:
Merge branch 'develop' into feature/GH-14030

pawelfras committed:
Merge branch 'develop' into feature/GH-14030

pawelfras committed:
build fixes

pawelfras committed:
CI fix

pawelfras committed:
Trigger Build

pawelfras committed:
Trigger Build

pawelfras committed:
Merge branch 'develop' into feature/GH-14030

pawelfras committed:
Merge branch 'develop' into feature/GH-14030

Zeyber committed:
fix: Fix mailto and tel generic links (#15429)

Closes: #15355

BenjaminEmiliani committed:
style: add background color to active tab in PDP (#15466)

BenjaminEmiliani committed:
fix: improve SR a11y on NVDA/JAWS for added to cart  (#15338)

Zeyber committed:
fix: Change generic link component properties access to protected (#15513)

morganm58 committed:
feature/GH-15328 (#15435)

fixed qty and total saved cart table alignment

Co-authored-by: RadhepS <34665674+RadhepS@users.noreply.github.com>

BenjaminEmiliani committed:
fix: determine error message for email update error (#15447)

BenjaminEmiliani committed:
fix: bind aria attributes to ng-select in checkout (#15398)

hackergil committed:
Adding EPD visualization to release steps. Also sorting libraries based on dependencies

morganm58 committed:
style: Updating theme colors (#15516)

Platonn committed:
feat: upgrade to Angular 13 (#14323)

Main changes:
- bump `@angular/` packages from v12.0 to v13.2
- bump `typescript` from v4.2 to v4.4
- bump relative packages (e.g. `@nguniversal`, `@ngrx`, `ng-select` etc.) to the highest version compatible with ng13.2
- bump `i18next` from v20 to v21
- bump css `bootstrap` from v4.3 to v4.6

Consequence changes:
- drop section `umdModuleIds` from `ng-package.json` files - it's not needed since ng13
- add "exports/sass" section in `package.json` files of libraries - needed since ng13 introduced new "Angular Package Format"
- bump required Node.js version in our repo to `^14.15.0 || >=16.10.0` - same as in ng13 (but excluding Node12 because on 2022-04-30 Node.js v12 will become EOL)
  - use Node 16 (instead of Node 12) in CI
- add eslint rules needed after bumping `@angular-eslint` and `eslint`
  - rename variables here and there to conform the `no-shadow` eslint rule
  - change comparisons `==` to `===` here and there to conform `eqeqeq` rule
- bump `jest`, bump `ts-jest` and add `jest-preset-angular` - needed since ng13 is shipped only as ES Modules (no CommonJS)
  - btw. renamed files `jest.ts` to `test-jest.js`
- configure `destroyAfterEach: false` for unit tests of existing libraries - since ng13 the default is `destroyAfterEach: true`
- use CSS `justify-content: 'flex-start',` instead of `'start'` to avoid to warning since ng13
- bump `semver` package used in our schematics - same as ng13
- import ES Module of `@angular/compiler` in our CommonJS migration-schematics via a workaround function `loadEsmModule()` - since ng13 `@angular/compiler` is shipped only as ES Module
- force downgrade of rxjs in the fresh app from v7 to v6  via installation-schematics  - needed since `ng new` in v13 installs rxjs@7, but Spartacus requires v6
- update `.browserslistrc` for `storefrontapp` project - same as in a fresh ng13 app
- add default value `'.'` of `@Input pageRoute` in `PaginationComponent` - needed since ng13 disables links with empty `routerLink`
- add simple unit tests in some libraries - after upgrading to ng13 the coverage was below the threshold for some libraries
- add new param `target` of `context.scheduleBuilder()` in `augmented-types.ts` builder - needed since ng13
- change NgRx `@Effect()` decorator to `createEffect()` function  - needed since ngrx@13
- update sufixes of plural i18next keys from `_plural` to `_other` - needed since i18next@21

closes #13916

dydome committed:
style: Change breadcrumb styles to wrap long crumbs (#8104)

pawelfras committed:
Merge branch 'develop' into feature/GH-14030

pawelfras committed:
Trigger Build

developpeurweb reviewed (approved):
Still works well on JAWS/NVDA/VO.
